scripts/imagebuild.sh

```diff
@@ -0,0 +1,51 @@
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+base_dir="$(cd "$(dirname "$0")/.." ; pwd)"
+dockerfile_dir="${base_dir}/images/federation-v2"
+cd "$base_dir/bin" || {
```

Comment:
I now realise that the actual check you intend to do is rather check the presence of actual binaries because u use them later. For this checking existence of `$base_dir/bin` directory  alone is insufficient. What if `$base_dir/bin` is there but no actual binaries in it.
I guess you should be doing 
`[ -f "$base_dir/bin/apiserver" ] || { echo "$base_dir/bin/apiserver not found" ; exit 1 ;}`
`[ -f "$base_dir/bin/controller-manager" ] || { echo "$base_dir/bin/controller-manager not found" ; exit 1 ;}`

Reply:
yep, that makes more sense.